src/wallet/rpcdump.cpp

```diff
@@ -1137,12 +1137,12 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
                 // range, or if the import result already has an error set, let
                 // the result stand unmodified. Otherwise replace the result
                 // with an error message.
-                if (GetImportTimestamp(request, now) - TIMESTAMP_WINDOW >= scannedRange->GetBlockTimeMax() || results.at(i).exists("error")) {
+                if (GetImportTimestamp(request, now) - TIMESTAMP_WINDOW > scanFailed->GetBlockTimeMax() || results.at(i).exists("error")) {
```

Comment:
> That's the part I'm not sure about: aren't the blocks at and after that point missing, instead of before that point?

Because it returns a pointer to the last block that couldn't be scanned,  it means that any blocks after the pointer were scanned successfully.

> After all it's the point where it failed in a linear forward scan?

The scan does keep going even if individual blocks couldn't be read. There might be a smarter way to handle this, but the behavior has been around a while. Only the error reporting is fairly new.